src/main/java/org/elasticsearch/transport/netty/NettyTransport.java

```diff
@@ -604,6 +605,13 @@ void exceptionCaught(ChannelHandlerContext ctx, ExceptionEvent e) throws Excepti
             // close the channel as safe measure, which will cause a node to be disconnected if relevant
             ctx.getChannel().close();
             disconnectFromNodeChannel(ctx.getChannel(), e.getCause());
+        } else if (e.getCause() instanceof SizeHeaderFrameDecoder.HttpOnTransportException) {
+            // in case we are able to return data, serialize the exception content and sent it back to the client
+            if (ctx.getChannel().isOpen()) {
+                ChannelBuffer buffer = ChannelBuffers.wrappedBuffer(e.getCause().getMessage().getBytes(Charsets.UTF_8));
+                ctx.getChannel().write(buffer);
+                ctx.getChannel().close();
```

Comment:
I fixed this to use the channel future as it was intended to be.. so when the future is done, `close` is called